src/Options/Applicative/Help/Types.hs

```diff
@@ -42,6 +53,44 @@ helpText (ParserHelp e s h u d b g f) =
 -- | Convert a help text to 'String'.
 renderHelp :: Int -> ParserHelp -> String
 renderHelp cols
-  = (`displayS` "")
-  . renderPretty 1.0 cols
+  = LT.unpack
+  . B.toLazyText
+  . renderAnsi
+  . layoutPretty (LayoutOptions (AvailablePerLine cols 1.0))
   . helpText
+
+renderAnsi :: SimpleDocStream Ann -> B.Builder
+renderAnsi
+  = renderCtxDecorated defaultStyle B.fromText renderPush renderPop
+  . alterAnnotationsS alter
+  where
+    alter :: Ann -> Maybe SetStyle
+    alter (AnnStyle setStyle) = Just setStyle
+    renderPush :: SetStyle -> SetStyle -> B.Builder
+    renderPush _ setStyle = B.fromString (styleToRawText setStyle)
+    renderPop :: SetStyle -> SetStyle -> B.Builder
+    renderPop setStyle _ = B.fromString (styleToRawText setStyle)
+
+renderCtxDecorated
+    :: Monoid ann
+    => ann
+    -> (Text -> B.Builder) -- ^ Render plain 'Text'
+    -> (ann -> ann -> B.Builder)  -- ^ How to render an annotation
+    -> (ann -> ann -> B.Builder)  -- ^ How to render the removed annotation
+    -> SimpleDocStream ann
+    -> B.Builder
+renderCtxDecorated topAnn toText push pop = go [topAnn]
```

Comment:
Is it possible to express `renderCtxDecorated` as `SimpleDocStream ann -> SimpleDocStream ()` pass, without touching `Data.Text.Builder` or anything else in `Data.Text` explicitly? I imagine you can replace `SAnn{Push,Pop}` with `SText` nodes, and then call `renderPretty` as before.

Otherwise (besides my primary interest to avoid `text` dependency) rendering `SimpleDocStream` on your own means a pretty fragile dependency on internals and semantics of `prettyprinter`, like the exhaustive list of constructors of `SimpleDocStream`, `panicUncaughtFail`, `panicInputNotFullyConsumed`, `panicUnpairedPop`, etc.

Reply:
This is a bit tricky.  I'm heading over to `prettyprinter` repo for advice.

Reply:
This code has been updated with @Bodigrim 's suggestion to avoid using `text` and it looks a bit simpler.